### PR TITLE
9.10.1 Release notes

### PR DESCRIPTION
**Pull request by lzdanski**
See [slack thread](https://astronomer.slack.com/archives/C026XPSJ0JK/p1713261535580039)

Includes changes for:

- [Load providers configuration when gunicorn workers start](https://github.com/apache/airflow/pull/38795)
- [Prevent large objects from being stored in the RTIF](https://github.com/apache/airflow/pull/38094)
- [Load consuming_dags attr eagerly before dataset listener](https://github.com/apache/airflow/pull/36247)
- [Add “return” statement to “yield” within a while loop in core triggers](https://github.com/apache/airflow/pull/38389)
- [Improve ExternalTaskSensor Async Implementation (#36916)](https://github.com/apache/airflow/pull/36916)
- [Disable edit/delete of monitoring dag in the UI](https://github.com/astronomer/astro-runtime/pull/1157)
- [Fix sqlparse vulnerability in versions <0.5.0](https://github.com/astronomer/astro-runtime/pull/1441)

### CODE REVIEW

File: astro/runtime-release-notes.md [hunk 1]
```diff
@@ -218,6 +218,28 @@ For more information about the major changes in this release, see the [Airflow b
...
+- Providers now load when Gunicorn workers start ([#38795](https://github.com/apache/airflow/pull/38795))
+- You can now customize the size of objects stored in the rendered `taskinstance` field ([#38094](https://github.com/apache/airflow/pull/38094))
+- Fixed an issue where the dataset listener could cause an error by closing the session prematurely ([#36247](https://github.com/apache/airflow/pull/36247))
+- Fixed an issue where core triggers were not exiting a `while` loop as expected ([#38389](https://github.com/apache/airflow/pull/38389))
+- Updated the behavior of the asynchornous implementation of `ExternalTaskSensor` to work more similarly to the synchronous implementation([#36916](https://github.com/apache/airflow/pull/36916))
+
+### Additional Improvements
+
+- You can now edit and delete monitoring DAGs in the Astro UI.
+- Updated the version of `sqlparse` to `0.5.0.
```
Comment (utkarsharma2) [inline astro/runtime-release-notes.md:237]:
```suggestion
- Updated the version of `sqlparse` to `0.5.0`.
```

File: astro/runtime-release-notes.md [hunk 2]
```diff
@@ -218,6 +218,28 @@ For more information about the major changes in this release, see the [Airflow b
...
+
+- Providers now load when Gunicorn workers start ([#38795](https://github.com/apache/airflow/pull/38795))
+- You can now customize the size of objects stored in the rendered `taskinstance` field ([#38094](https://github.com/apache/airflow/pull/38094))
+- Fixed an issue where the dataset listener could cause an error by closing the session prematurely ([#36247](https://github.com/apache/airflow/pull/36247))
+- Fixed an issue where core triggers were not exiting a `while` loop as expected ([#38389](https://github.com/apache/airflow/pull/38389))
+- Updated the behavior of the asynchornous implementation of `ExternalTaskSensor` to work more similarly to the synchronous implementation([#36916](https://github.com/apache/airflow/pull/36916))
+
+### Additional Improvements
+
+- You can now edit and delete monitoring DAGs in the Astro UI.
```
Comment (utkarsharma2) [inline astro/runtime-release-notes.md:236]:
```suggestion
- Fix for missing `Cluster Activity` tab ([#1157] (https://github.com/astronomer/astro-runtime/pull/1157))
```

**Review by utkarsharma2 — APPROVED** [review]

File: astro/runtime-release-notes.md (hunk 2 above)
Comment (utkarsharma2) [reply astro/runtime-release-notes.md:236]: Its actually a bug fix.

Comment (lzdanski) [reply astro/runtime-release-notes.md:236]: There's some logic about what kind of changes go where and when we link to github - edited your suggestion and added it to the Bug fixes section now (line 241)

File: astro/runtime-release-notes.md [hunk 3]
```diff
@@ -218,6 +218,28 @@ For more information about the major changes in this release, see the [Airflow b
...
+- Updated the version of `sqlparse` to `0.5.0`.
+
+### Bug fixes
+
+- Fixed a bug where the `Cluster Activity` tab was missing.
```
Comment (jwitz) [inline astro/runtime-release-notes.md:241]:
```suggestion
- Fixed a bug where the **Cluster Activity** tab was missing from the Airflow UI.
```

**Pull request merged**